datalake/src/main/java/com/sequenceiq/datalake/flow/create/SdxCreateFlowConfig.java

```diff
@@ -24,24 +24,35 @@
 
     private static final List<Transition<SdxCreateState, SdxCreateEvent>> TRANSITIONS = new Builder<SdxCreateState, SdxCreateEvent>()
             .defaultFailureEvent(SDX_CREATE_FAILED_EVENT)
+
             .from(INIT_STATE)
-            .to(SDX_CREATION_WAIT_ENV_STATE)
+            .to(SDX_CREATION_WAIT_RDS_STATE)
             .event(SdxCreateEvent.ENV_WAIT_EVENT).defaultFailureEvent()
-            .from(SDX_CREATION_WAIT_ENV_STATE)
+
+            .from(INIT_STATE)
             .to(SDX_CREATION_WAIT_RDS_STATE)
-            .event(SdxCreateEvent.ENV_WAIT_SUCCESS_EVENT).defaultFailureEvent()
+            .event(SdxCreateEvent.RDS_WAIT_EVENT).defaultFailureEvent()
+
             .from(SDX_CREATION_WAIT_RDS_STATE)
-            .to(SDX_CREATION_START_STATE)
+            .to(SDX_CREATION_WAIT_ENV_STATE)
             .event(SdxCreateEvent.RDS_WAIT_SUCCESS_EVENT).defaultFailureEvent()
+
+            .from(SDX_CREATION_WAIT_ENV_STATE)
+            .to(SDX_CREATION_START_STATE)
+            .event(SdxCreateEvent.ENV_WAIT_SUCCESS_EVENT).defaultFailureEvent()
```

Comment:
I think it's possible that we have the old version running, and it's in `SDX_CREATION_WAIT_ENV_STATE`. We shut it down and fire up this new version. When it's finished with the `SDX_CREATION_WAIT_ENV_STATE` the new version will move to `SDX_CREATION_START_STATE` and will skip `SDX_CREATION_WAIT_RDS_STATE`. Could you check?